projects/ngrx.io/content/guide/store/recipes/adding-ngrx-to-existing-apps.md

```diff
@@ -0,0 +1,499 @@
+# Adding NgRx to existing Angular applications
+
+## How do apps without NgRx typically function?
+
+Most of the applications out there do not use NgRx. But most applications follow the typical patterns and solve common problems, which are:
+
+1. Retrieving data from a remote API
+2. Saving that data in a component state (say, on some property like `userList`) 
+3. Display that data in the UI
+4. Make alterations of the data in the backend (add, update or delete entries)
+5. Again, display the results of those actions in the UI (success messages, updated lists of data, and so on)
+
+Usually to do that, developers write services which inject the Angular's `HttpClient`, write methods to retrieve or modify specific parts of the data, then inject those services in the components and use them, tightly coupled with the components. Let's see an example of such a "conventional" component:
+
+<code-example header="to-do-list.component.ts">
+@Component({
+    selector: 'app-to-do-list',
+    template: `
+      &lt;div&gt;
+        &lt;ul&gt;
+          &lt;li *ngFor="let todo of todos"&gt;
+            &lt;span&gt;{{ todo.text }}&lt;/span&gt;
+            &lt;button (click)="removeTodo(todo.id)"&gt;X&lt;/button&gt;
+          &lt;/li&gt;
+        &lt;/ul&gt;
+        &lt;div&gt;
+          &lt;input #todoName placeholder="Add new To-Do Item"/&gt;
+          &lt;button (click)="addToDo(#todoName.value)"&gt;Save To-Do Item&lt;/button&gt;
+      &lt;/div&gt;
+    `
+})
+export class ToDoListComponent implements OnInit {
```

Comment:
@markostanimirovic I can do that, but it will be a fairly large effort for me at the moment (being busy at my job). Do you think it is warranted? For me, either of the examples made sense, I think a "To Do" item is as tangible as a "Book". Although if you think it will be good to have another example instead of this, I can pull through

Reply:
Agree, CRUD examples are similar for todo or another type of entity, but I'd rather replace "todo" to make the example closer to the real-world situation. Thanks a lot for your effort, Armen! It's not urgent, we can postpone this PR when you have more time. 🙂

Btw, I noticed several different namings for "todo" in the guide: "To-Do", "ToDo", "Todo". "Book" will solve that. 😅